packages/utils/src/common/delay/index.ts

```diff
@@ -1,7 +1,12 @@
-export const delay = (time: number): Promise<void> => {
+export const delay = (time: number, callback?: () => void): Promise<void> => {
   return new Promise<void>((resolve, reject) => {
-    if (isNaN(time) || time < 0) reject(new Error('Invalid time value'));
```

Comment:
@ssi02014 엇 제가 작성했던 함수를 디벨롭해주셨군요! 너무 감사합니다! 👍 

한가지 의견드리고자 하는 점은 작성해주신 코드를 보면 NaN과 음수의 케이스에 더불어 Infinity와 -Infinity까지도 체크해주신 것으로 확인했는데요, 실제 사용성측면에서 Infinity 혹은 -Infinity가 들어올 케이스는 거의 없을듯하여 과연 필요한 부분일까 한번 생각하게 되었습니다!

단순하게 반대의 의견을 드리는 것은 아니고, 위 케이스를 완벽하게 커버하겠다면 Number.isInteger에서의 예외케이스에 포함되어있는 소수점도 체크하여 예외케이스에 넣는 방법으로 가거나, 실제 사용성을 고려해서 기존의 NaN + 음수까지만 체크하는 방식 두가지 중 한가지 방향으로 정하면 어떨지 의견드려봅니다!

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/isInteger#using_isinteger

```typescript
// 아래와 같은 유틸함수를 활용하면 소수점케이스는 전부 제거할 수 있을 것으로 보입니다!
function hasDecimal(number) {
  return number % 1 !== 0;
}
```

Reply:
의견 감사합니다!

저는 추가적인 함수를 작성 할 필요없이 잘 설계된 js에서 제공하는 built-in 함수를 활용해 거의 모든 예외 케이스를 모두 대응할 수 있으면 좋을 것 같은데 어떻게 생각하실까요!?

Number.isInterger 함수를 사용한 이유는 제가 `소수`, `NaN`, `Infinity`와 같은 케이스에서 유효성 검사도 있지만 궁극적으로 number 타입에 대한 대부분의 유효성 검사를 대부분을 진행해주기 때문입니다!

```ts
Number.isInteger(0); // true
Number.isInteger(1); // true
Number.isInteger(-100000); // true
Number.isInteger(99999999999999999999999); // true

Number.isInteger(0.1); // false
Number.isInteger(Math.PI); // false

Number.isInteger(NaN); // false
Number.isInteger(Infinity); // false
Number.isInteger(-Infinity); // false
Number.isInteger("10"); // false
Number.isInteger(true); // false
Number.isInteger(false); // false
Number.isInteger([1]); // false
```

Reply:
@ssi02014 의도하신 부분에 대해 자세히 설명해주셔서 감사합니다! 👍 
말씀주신 목적이라면 기존의 isNaN을 통한 validation보다 훨씬 좋겠네요! 별도 함수 추가 없이 isInteger로 validation을 진행해도 좋아보입니다! 🙇

Reply:
@Sangminnn  사실 `타입스크립트` 사용자라면 `타입` + `hasDecimal`로 대부분의 문제를 해결 할 수 있지만 조금 더 광범위한 유효성 검사를 진행 할 수 있는 방향으로 고려해봤던 것 같습니다! 🤔

```ts
Number.isInteger(5.0000000000000001); // true, because of loss of precision
Number.isInteger(4500000000000000.1); // true, because of loss of precision
```
헉 위와 같은 케이스는 제가 고려해보지 못했네요 🥲 너무 좋은 참고가 됐습니다 🙇‍♂️